Assets/BossRoom/Scripts/Client/UI/PopupPanel.cs

```diff
@@ -13,9 +13,16 @@ public class PopupPanel : MonoBehaviour
         TextMeshProUGUI m_TitleText;
         [SerializeField]
         TextMeshProUGUI m_MainText;
+        [SerializeField]
+        GameObject m_ConfirmButton;
+        [SerializeField]
+        GameObject m_LoadingImage;
 
         bool m_IsPopupShown;
 
+        long m_DisplayedPopupId = -1;
```

Comment:
shouldn't resetState reset this as well?
(that's why I hate "reset" functions, that's really against OOP. You usually just want to destroy your objects and create new ones)

Reply:
I agree, I'm not a fan of those as well. For this PR I'll simply add that to ResetState, but I'll definitely rework this in my PR implementing stacking

Reply:
In its current state, it already solves some of the issues with this, but it can be better I think